configuration/ConfigHandler.cpp

```diff
@@ -0,0 +1,50 @@
+#include "ConfigHandler.hpp"
+#include <iostream>
+#include <fstream>
+#include "../yaml-cpp/include/yaml-cpp/yaml.h"
+
+Configuration::ConfigHandler::ConfigHandler(std::string filePath)
+{
+    this->getConfigFile(filePath);
+}
+
+Configuration::ConfigHandler::~ConfigHandler(){}
+
+void Configuration::ConfigHandler::getConfigFile(std::string configFile)
+{
+    YAML::Node config = YAML::LoadFile(configFile);
+
+    const std::string ip = config["network"]["ip"].as<std::string>();
```

Comment:
There is no error handling whatsover, please check the return values of the config access as the fields could not exist, as right now this would just trigger a segmentation fault.